### PR TITLE
Add rate limit for creating backup group to increase stability

### PR DESCRIPTION
**Pull request by mwojtyczka**
* Fix for issue [495](https://github.com/databrickslabs/ucx/issues/495) 
* Updated rate limits as per: https://learn.microsoft.com/en-us/azure/databricks/resources/limits#--api-rate-limits
* Fixed end to end test

### CODE REVIEW

File: src/databricks/labs/ucx/workspace_access/groups.py [hunk 1]
```diff
@@ -286,14 +287,14 @@ def _replace_group(self, migration_info: MigrationGroupInfo):
...
-    @rate_limited(max_requests=5)
+    @rate_limited(max_requests=35)
     def _delete_workspace_group(self, ws_group: iam.Group) -> None:
         logger.info(f"Deleting the workspace-level group {ws_group.display_name} with id {ws_group.id}")
         self._ws.groups.delete(id=ws_group.id)
         logger.info(f"Workspace-level group {ws_group.display_name} with id {ws_group.id} was deleted")
 
     @retried(on=[DatabricksError])
-    @rate_limited(max_requests=10)
+    @rate_limited(max_requests=5)
```
[the remark this answers is not in the document]
Comment (mwojtyczka) [reply src/databricks/labs/ucx/workspace_access/groups.py:297]: Based on this list: https://learn.microsoft.com/en-us/azure/databricks/resources/limits#--api-rate-limits

[the remark this answers is not in the document]
Comment (mwojtyczka) [reply src/databricks/labs/ucx/workspace_access/groups.py:297]: but this is even wrong, once is request per second the other per minute

[the remark this answers is not in the document]
Comment (mwojtyczka) [reply src/databricks/labs/ucx/workspace_access/groups.py:297]: I updated the rate limits as some of them were actually per minute not seconds

**Review by nfx — APPROVED** [review]
okay to merge if integration tests pass

**Review by william-conti — APPROVED** [review]

**Pull request merged**